src/wasm-stack.h

```diff
@@ -251,7 +260,22 @@ void BinaryenIRWriter<SubType>::visit(Expression* curr) {
     // `curr` is not reachable, so don't emit it.
     return;
   }
-  emitDebugLocation(curr);
+
+  if (func != nullptr) {
+    const auto& debugLocationIterator = func->debugLocations.find(curr);
+    if (debugLocationIterator != func->debugLocations.cend()) {
+      if (lastDebugLocation != debugLocationIterator->second ||
```

Comment:
for the function debug info emit, we add the depth logic, if the parent or siblings instruction already inserted the debuginfo, the instruction will not emit the debug info in source map generation, it will not impact the debug info reload from source map (children and siblings instructions will also get the debug info when load them from source map to memory structures), just optimize the generation of source map.
@kripken